Xamarin.Forms.Core.UnitTests/ButtonUnitTest.cs

```diff
@@ -323,5 +324,32 @@ private void AssertButtonContentLayoutsEqual(Button.ButtonContentLayout layout1,
 			Assert.AreEqual(layout1.Position, bcl.Position);
 			Assert.AreEqual(layout1.Spacing, bcl.Spacing);
 		}
+
+		[Test]
+		public void ButtonCornerRadiusSetInExplicitStyle()
+		{
+			var xaml = @"<?xml version=""1.0"" encoding=""utf-8"" ?>
+			             <ContentPage xmlns=""http://xamarin.com/schemas/2014/forms""
+			                          xmlns:x=""http://schemas.microsoft.com/winfx/2009/xaml""
+			                          x:Class=""Xamarin.Forms.Core.UnitTests.MyTestContentPage"">
+			                 <ContentPage.Resources>
+			                     <Style TargetType=""Button"">
+			                         <Setter Property=""TextColor"" Value=""White""></Setter>
+			                     </Style>
+			                     <Style x:Key=""ButtonStyle"" TargetType=""Button"">
+			                         <Setter Property=""CornerRadius"" Value=""0""></Setter>
+			                         <Setter Property=""TextColor"" Value=""Yellow""></Setter>
+			                     </Style>
+			                 </ContentPage.Resources>
+			                     <Button x:Name=""Button0"" Style=""{StaticResource ButtonStyle}"" />
+			             </ContentPage>";
+
+			var contentPage = new MyTestContentView();
```

Comment:
Why not a regular `ContentView`?

Reply:
No special reason. I changed it now. I guess it looks cleaner to simply use a ContentView?

Reply:
If there is no special reason to create an inheritance we might as well just use the base class indeed.